src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs

```diff
@@ -165,6 +167,24 @@ public virtual async Task<ImmutableList<ChatInfo>> List(ChatListOrder order, Can
         return result;
     }
 
+    private async Task<string> GetLastEntryTextContent(ChatEntry? entry, CancellationToken cancellationToken)
+    {
+        if (entry is null)
+            return "";
+
+        if (!entry.IsSystemEntry)
+            return entry.GetContentOrDescription();
+
+        var membersChanged = entry.SystemEntry?.MembersChanged;
+        if (membersChanged != null) {
+            var author = await Authors.Get(Session, entry.ChatId, membersChanged.AuthorId, cancellationToken).ConfigureAwait(false);
+            var action = membersChanged.HasLeft ? "left" : "joined";
+            return $"{author.Avatar.Name} {action} the group";
+        }
+
+        throw StandardError.Constraint("Unexpected system entry type");
```

Comment:
I think it's too strict behavior. Do not throw exception here.
Better to have empty line rather than failed app.

Reply:
done